apps/ledger-live-mobile/src/newArch/features/FirmwareUpdate/components/UpdateBanner/useUpdateBannerViewModel.test.tsx

```diff
@@ -0,0 +1,232 @@
+import { act, renderHook } from "@testing-library/react-native";
+import { useUpdateBannerViewModel } from "./useUpdateBannerViewModel";
+
+/** Brace yourselves for the mocks ... */
+
+jest.mock("@react-navigation/native", () => ({
+  ...jest.requireActual("@react-navigation/native"),
+  useRoute: jest.fn(),
+  useNavigation: jest.fn(),
+}));
+jest.mock("react-redux", () => ({
+  ...jest.requireActual("react-redux"),
+  useSelector: jest.fn().mockImplementation((selector: () => unknown) => selector()),
+}));
+
+jest.mock("~/reducers/settings", () => ({
+  ...jest.requireActual("~/reducers/settings"),
+  lastSeenDeviceSelector: jest.fn(),
+  lastConnectedDeviceSelector: jest.fn(),
+  hasCompletedOnboardingSelector: jest.fn(),
+}));
+const { lastConnectedDeviceSelector, hasCompletedOnboardingSelector } =
+  jest.requireMock("~/reducers/settings");
+
+jest.mock("~/reducers/appstate", () => ({
+  ...jest.requireActual("~/reducers/appstate"),
+  hasConnectedDeviceSelector: jest.fn(),
+}));
+const { hasConnectedDeviceSelector } = jest.requireMock("~/reducers/appstate");
+
+jest.mock("@ledgerhq/live-common/device/hooks/useLatestFirmware", () => ({
+  ...jest.requireActual("@ledgerhq/live-common/device/hooks/useLatestFirmware"),
+  useLatestFirmware: jest.fn(),
+}));
+const { useLatestFirmware } = jest.requireMock(
+  "@ledgerhq/live-common/device/hooks/useLatestFirmware",
+);
+
+jest.mock("../../utils/isFirmwareUpdateSupported", () => ({
+  ...jest.requireActual("../../utils/isFirmwareUpdateSupported"),
+  isOldFirmwareUpdateUxSupported: jest.fn().mockImplementation(() => ({
+    updateSupported: true,
+    updateSupportedButDeviceNotWired: false,
+  })),
+  isNewFirmwareUpdateUxSupported: jest.fn().mockImplementation(() => ({ updateSupported: true })),
+}));
+const { isOldFirmwareUpdateUxSupported, isNewFirmwareUpdateUxSupported } = jest.requireMock(
+  "../../utils/isFirmwareUpdateSupported",
+);
+
+jest.mock("../../utils/navigateToFirmwareUpdateFlow", () => ({
+  ...jest.requireActual("../../utils/navigateToFirmwareUpdateFlow"),
+  navigateToFirmwareUpdateFlow: jest.fn(),
+}));
+
+jest.requireMock("../../utils/navigateToFirmwareUpdateFlow");
+const navigateToFwUpdateFlowSpy = jest.spyOn(
+  jest.requireMock("../../utils/navigateToFirmwareUpdateFlow"),
+  "navigateToFirmwareUpdateFlow",
+);
```

Comment:
I think that you should cover this hook through an integration test. You mock too much stuff here. We can have a look together if you want.